spec/gogo/x/username/codec.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package username;
+
+import "codec.proto";
+import "gogoproto/gogo.proto";
+
+// Token model represents a username mapping to an address together with all
+// metadata.
+//
+// Each Token model is stored using the username as the key. This guarantee
+// that the name is unique. Username is a combination of a name and a domain.
+// The format is <name>*<domain>
+//
+// Each token points to a blockchain and an address on that blockchain. Both
+// blockchain ID and address are an arbitrary string as we do not want to limit
+// ourselves to certain patterns.
+message Token {
+  weave.Metadata metadata = 1;
+  // Target specifies where this username token points to.
+  Location target = 2 [(gogoproto.nullable) = false];
```

Comment:
req: a single username can point to multiple chain addresses

Reply:
Implemented in https://github.com/iov-one/weave/pull/771/commits/288a34c4bafd8780ebedba1352d29310f477e28b